plugins/permission-backend-module-opa-wrapper/src/opa-client/opaClient.ts

```diff
@@ -25,21 +26,27 @@ export class OpaClient {
     this.opaEntryPoint = config.getOptionalString(
       'opaClient.policies.permissions.entrypoint',
     );
-    this.logger = logger;
     this.opaBaseUrl = config.getOptionalString('opaClient.baseUrl');
+    this.opaPolicyFallback = config.getOptionalString(
+      'opaClient.policies.permissions.policyFallback',
+    );
+    this.logger = logger;
   }
 
   /**
    * Evaluates a policy against a given input.
    * @param input - The input to evaluate the policy against.
    * @param opaEntryPoint - The entry point into the OPA policy to use. You can optionally provide the entry point here, otherwise it will be taken from the app-config.
+   * @param opaPolicyFallback - The fallback policy to use when the OPA server is unavailable or unresponsive. You can optionally provide the fallback policy here, otherwise it will be taken from the app-config.
    */
   async evaluatePolicy(
     input: PolicyEvaluationInput,
     opaEntryPoint?: string,
+    opaPolicyFallback?: string,
   ): Promise<PolicyEvaluationResult> {
     const setEntryPoint = opaEntryPoint ?? this.opaEntryPoint;
     const opaBaseUrl = this.opaBaseUrl;
+    const policyFallback = opaPolicyFallback ?? this.opaPolicyFallback;
```

Comment:
I think we can simplify this method a little bit, we don't need to pass the `opaPolicyFallback` to it and we can simply do the below, as its already being set during the instantiation of the class

```ts
const policyFallback = this.opaPolicyFallback;
```

WDYT?

Reply:
I think that's a good idea, and I had thought about doing that. However, the thing is it makes it harder to test. Right now the Config has a mocked implementation, which makes it pretty hard to test different config values (as I have to alternate between 'allow' and 'deny'). I'm changing the tests a bit to make your change possible, LMK what you think. I'll link the commit once I've finished it

Reply:
https://github.com/Parsifal-M/backstage-opa-plugins/pull/220/commits/3ba93512564d416f953f8b0584b02ff15e97c1d1

I dropped the ConfigReader.getOptionalString mock implementation, and replaced with just a base config object read by ConfigReader set in the beforeAll() of the jest tests. This makes it easier to test a slightly modified config.